extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java

```diff
@@ -91,7 +91,7 @@ InfinispanPropertiesBuildItem setup(ApplicationArchivesBuildItem applicationArch
         // Enable SSL support by default
         sslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(FeatureBuildItem.INFINISPAN_CLIENT));
 
-        ClassLoader cl = Thread.currentThread().getContextClassLoader();
```

Comment:
This should also use loadsApplicationClasses and just use the TCCL (not the fault of this PR, but while we are fixing these issues might as well get this one too).

I know it seems like a bit of a corner case (the wrong TCCL being set during class loading), but this did actually cause real issues in the early days of WildFly so it can be a problem.

Reply:
That makes sense. PR updated